src/optimizers/adam.cpp

```diff
@@ -206,7 +217,8 @@ void adam<TensorDataType>::step_compute_cpu(AbsDistMatrixType& values,
       auto& m2 = moment2_buffer[i];
       m1 = m_beta1 * m1 + (one - m_beta1) * g;
       m2 = m_beta2 * m2 + (one - m_beta2) * g * g;
-      x -= correction * m1 / (El::Sqrt(m2) + m_eps);
+      x -=
+        correction * (m1 / (El::Sqrt(m2) + m_eps) + m_adamw_weight_decay * x);
```

Comment:
I think this may apply the AdamW weight decay wrong. This scales the weight decay term by `correction`, which is `learning_rate * sqrt(1-beta2) / (1 - beta1)`, and so scales the weight decay by the correction factor for the moments.

I believe it should instead scale the weight decay term only by the learning rate:
```
x -= correction * m1 / (El::Sqrt(m2) + m_eps) + learning_rate * m_adamw_weight_decay * x
```

Please double-check this.

Reply:
Fixed.